### PR TITLE
Create user

### LINKED ISSUE #2: Create New User
Form with the following fields:
(form_with)
- [x] First Name (string)
- [x] Last Name (string)
- [x] Email (string)

### PR DESCRIPTION
**Pull request by jdcarey128**
This PR: 
- Creates Users table migration 
- Defines User factory
- Tests User model 
- Creates GraphQL UserType and createUser mutation 
- Tests createUser mutation with edge cases 
- Adds filter for simpleCov
- Add simple PR template

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<!--- Provide a general summary of your changes in the Title above -->
```
Comment (dcoleman21) [inline .github/pull_request_template.md:1]: Thanks for adding this PR template

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,4 @@
 
 # Ignore master key for decrypting credentials and more.
 /config/master.key
+coverage
```
Comment (dcoleman21) [inline .gitignore:24]: Good call here.

File: app/graphql/mutations/create_user.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+module Mutations 
+  class CreateUser < Mutations::BaseMutation
```
Comment (dcoleman21) [inline app/graphql/mutations/create_user.rb:2]: This looks very straight forward

File: app/graphql/types/query_type.rb [hunk 4]
```diff
@@ -4,14 +4,5 @@ class QueryType < Types::BaseObject
     include GraphQL::Types::Relay::HasNodeField
```
Comment (dcoleman21) [inline app/graphql/types/query_type.rb:4]: What is this doing?

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # This file is copied to spec/ when you run 'rails generate rspec:install'
 require 'simplecov'
 SimpleCov.start
+SimpleCov.add_filter ['app/controllers/graphql_controller.rb', 'app/graphql/garden_share_be_schema.rb', 'spec/', 'config/']
```
Comment (dcoleman21) [inline spec/rails_helper.rb:4]: What is this line actually doing?

**Review by dcoleman21 — APPROVED** [review]
This all seems straight forward. Thank you for setting this up. I can use this as a template of sorts for creating others.

File: .github/pull_request_template.md (hunk 1 above)
Comment (Callbritton) [reply .github/pull_request_template.md:1]: Indeed! Looks great.

File: app/graphql/mutations/create_user.rb (hunk 3 above)
Comment (Callbritton) [reply app/graphql/mutations/create_user.rb:2]: As Dani said, this looks very familiar!

File: app/graphql/types/mutation_type.rb [hunk 6]
```diff
@@ -1,10 +1,5 @@
 module Types
   class MutationType < Types::BaseObject
-    # TODO: remove me
-    field :test_field, String, null: false,
-      description: "An example field added by the generator"
-    def test_field
-      "Hello World"
-    end
+    field :create_user, mutation: Mutations::CreateUser
```
Comment (Callbritton) [inline app/graphql/types/mutation_type.rb:3]: It's great to see how this one-line is hand rolled for future reference.

**Review by Callbritton — APPROVED** [review]
